garak/generators/nim.py

```diff
@@ -174,4 +175,94 @@ def _prepare_prompt(self, prompt):
         return text
 
 
+class NVMultimodal(RestGenerator):
+    """Wrapper for text + image / audio to text NIMs. Expects NIM_API_KEY environment variable.
+
+    Expects keys to be a dict with keys 'text' (required), and 'image' or 'audio' (optional).
+    Message is sent with 'role' and 'content' where content is structured as text
+    followed by <img> and/or <audio> tags ala https://build.nvidia.com/microsoft/phi-4-multimodal-instruct
+    """
+
+    DEFAULT_PARAMS = RestGenerator.DEFAULT_PARAMS | {
+        "headers": {
+            "Authorization": "Bearer $KEY",
+            "Accept": "application/json",
+        },
+        "req_template_json_object": """{
+        "model": "$NAME",
+        "messages": [
+            {
+                "role": "user",
+                "content": "$INPUT",
+            }
+        ],
+        "max_tokens": 512,
+        "temperature": 0.10,
+        "top_p": 0.70,
+        "stream": False,
+    }""",
+    }
+
+    ENV_VAR = "NIM_API_KEY"
+    generator_family_name = "NVMultimodal"
+    modality = {"in": {"text", "image", "audio"}, "out": {"text"}}
+    uri = "https://integrate.api.nvidia.com/v1/chat/completions"
+
+    def __init__(self, name, config_root=_config):
+        super.__init__(self.uri, config_root=config_root)
+        self.name = name
+
+    def _populate_template(
+        self, template: str, text: str, json_escape_key: bool = False
+    ) -> str:
+        output = template
+
+        if "$KEY" in template:
+            if self.api_key is None:
+                raise APIKeyMissingError(
+                    f"Template requires an API key but {self.key_env_var} env var isn't set"
+                )
```

Comment:
This should already be handled by `_validate_env_var()` from the `RestGenerator`.

Reply:
Recoding API key checking comes up not infrequently in PRs. This is a signal that we could all gain time from making routes to understanding the intended way, clearer. Recommended resolution:

* documenting this functionality itself explicitly
* adding a clear note in the base generator docs pointing to the method's docs
* covering it in the rst tutorial on generator writing
* specifically, because it's read & used so much, pointing to it in the RestGenerator docs

Reply:
Agreed, we should add some targeted documentation about consolidating any environment variable requirements for plugins in `_validate_env_var()` to ensure consistent management of required value collection.